### DISCUSSION TITLE
Subprotocols in 4.3.2

### DISCUSSION THREAD

**Discussion by B-R-Bender** [socketio/socket.io#4244] [Q&A] [answered]
Hello!
Is there a way to specify a list of subprotocols for ws transport while creating a socket?

something like

`const socket = io("wss://localhost:3500", {
    protocols: ["customProtocol"]
});`

Thank you!

**Comment by B-R-Bender** [comment]
looks like it's missing in 4.x (at least what I can find from [docs](https://socket.io/docs/v4/client-options/)) but was present in 3.x [here](https://socket.io/docs/v3/client-api/)
could you guys shed some light on this?

**Comment by B-R-Bender** [comment, marked as answer]
Alright, I've got it working, though it's not stated in the docs.
First you need to specify a transport

```
const socket = io("wss://localhost:3500", {
    transports: ["websocket"],
    protocols: ["custom-protocol"]
});
```

than on the server side you can get it with
```
io.on("connection", (socket) => {
  console.log(socket.client.conn.transport.socket.protocol);
});
```

**Reply by darrachequesne** [reply]
Hi! You are absolutely right, the option was indeed missing from the documentation.

It should be fixed now: https://socket.io/docs/v4/client-options/#protocols
